src/Metadata/DateTimeOptions.php

```diff
@@ -20,17 +20,18 @@ final class DateTimeOptions implements \JsonSerializable
     private $zone;
 
     /**
-     * Use if a different format should be used for parsing dates than for generating dates.
+     * Use if different formats should be used for parsing dates than for generating dates.
      *
-     * @var string|null
+     * @var string[]|null
      */
-    private $deserializeFormat;
+    private $deserializeFormats;
 
-    public function __construct(?string $format, ?string $zone, ?string $deserializeFormat)
+    public function __construct(?string $format, ?string $zone, ?string $deserializeFormat, ?array $allDeserializeFormats = [])
     {
         $this->format = $format;
         $this->zone = $zone;
-        $this->deserializeFormat = $deserializeFormat;
+        $deserializeFormat = \is_string($deserializeFormat) ? [$deserializeFormat] : $deserializeFormat;
```

Comment:
this is the same but less cognitive effort to understand, right?
```suggestion
        $deserializeFormat = \is_string($deserializeFormat) ? [$deserializeFormat] : null;
```